src/crafting.h

```diff
@@ -52,6 +52,7 @@ struct recipe {
     std::map<skill_id, int> required_skills;
     bool reversible; // can the item be disassembled?
     bool autolearn; // do we learn it just by leveling skills?
+    bool uncraft_single_charge; // recipe is to uncraft one charge of an item with multiple default charges
```

Comment:
Big whoops on that, I'm surprised my testing worked. I've fixed the initialisation in crafting.cpp now, and its working as expected for me with uncraft_single_charge set true or false in the json.